swan-lake/development-tutorials/java-interoperability/call-java-code-from-ballerina.md

```diff
@@ -1,31 +1,40 @@
+intro: Ballerina offers a straightforward way to call the existing Java code from Ballerina. Although Ballerina is not designed to be a JVM language, the current implementation, which targets the JVM, aka jBallerina, provides Java interoperability by adhering to the Ballerina language semantics.
 ---
 
-## Write Ballerina bindings
+## Ballerina bindings to Java code
+
+You can write Ballerina code (Ballerina bindings) that lets you call the corresponding Java API as illustrated in the diagram below.
+
+<img src="/learn/images/ballerina-interop-diagram-v1.png" alt="Ballerina bindings to Java code" width="300" height="450" style='width:auto !important; padding-top:20px; padding-bottom:20px;'>
 
-The `bindgen` is a CLI tool that eases the process of generating Ballerina bindings for given Java APIs. This guide shows you how to use it. For more details on how the `bindgen` tool works, see [Bindgen Tool](/learn/java-interoperability-guide/the-bindgen-tool/).
+## Ballerina FFI
 
-The [Package layout](/learn/package-references/#package-layout) explains how to package Java libraries (JAR files) with Ballerina programs to produce self-contained executable programs. When you generate bindings for a Java library using the Bindgen Tool, this part is already handled.
+The [Ballerina FFI](/learn/ballerina-ffi) explains how to write the Ballerina bindings manually. Learning to write them manually helps you to understand the inner workings of calling Java from Ballerina and how to customize the bindings generated by [the Bindgen tool](#the-bindgen-tool).
 
-The [Ballerina FFI](/learn/java-interoperability-guide/ballerina-ffi/) explains how to write these bindings manually. It is also a useful section for those who want to understand the inner workings of calling Java from Ballerina and for those who want to customize the bindings generated by the Bindgen Tool.
+## The Bindgen tool
 
-## Use the SnakeYAML Java library in Ballerina
-SnakeYAML is a YAML parser for Java. This section describes how to use this library to parse a YAML document using Ballerina. 
+Alternatively, you can use the [Bindgen tool](/learn/the-bindgen-tool/) to generate these bindings automatically while eliminating the need for understanding the Ballerina FFI layer. It is a CLI tool that eases the process of generating Ballerina bindings for given Java APIs. For more details on how the Bindgen tool works, see [The Bindgen tool](/learn/the-bindgen-tool/). It also explains how to customize the bindings generated by the Bindgen tool.
+
+## Write Ballerina bindings
 
-You develop a Ballerina program that parses the given YAML file and writes the content to the standard output.
+The [Package layout](/learn/package-references/#package-layout) explains how to package Java libraries (JAR files) with Ballerina programs to produce self-contained executable programs. When you generate bindings for a Java library using the Bindgen tool, this part is already handled.
 
-Let's get started.
+## Example
```

Comment:
Shouldn't this be and h3 title under `Write Ballerina bindings`?

Reply:
Thought it is fine to have this as an H2 title as this is one of the main highlights of the page. Changed the title `Use the SnakeYAML Java library in Ballerina`, which is the existing one as it is more meaningful.